rastervision_core/rastervision/core/utils/stac.py

```diff
@@ -28,32 +32,73 @@ def my_write_method(uri, txt):
         else:
             STAC_IO.default_write_text_method(uri, txt)
 
-    STAC_IO.read_text_method = my_read_method
-    STAC_IO.write_text_method = my_write_method
+    STAC_IO.read_text_method = read_method
+    STAC_IO.write_text_method = write_method
+
+
+def parse_stac(stac_uri: str) -> dict:
+    """Parse a STAC catalog JSON file to extract label URI, images URIs,
+    and AOIs.
 
+    Args:
+        stac_uri (str): Path to the STAC catalog JSON file.
 
-def parse_stac(stac_uri):
-    setup_stac_s3()
+    Returns:
+        dict: A dict with keys: "labels_uri", "labels_bbox", "image_uris",
+        and "aoi_geometry".
+    """
+    setup_stac_io()
     cat = Catalog.from_file(stac_uri)
     cat.make_all_asset_hrefs_absolute()
-    labels_uri = None
-    geotiff_uris = []
-    for item in cat.get_all_items():
-        if isinstance(item, LabelItem):
-            labels_uri = list(item.assets.values())[0].href
-            labels_box = box(*item.bbox)
-
-    # only use geotiffs that intersect with bbox of labels
-    for item in cat.get_all_items():
-        if not isinstance(item, LabelItem):
-            geotiff_uri = list(item.assets.values())[0].href
-            geotiff_box = box(*item.bbox)
-            if labels_box.intersects(geotiff_box):
-                geotiff_uri = geotiff_uri.replace('%7C', '|')
-                geotiff_uris.append(geotiff_uri)
-
-    if not labels_uri:
-        raise ValueError('Unable to read labels URI from STAC.')
-    if not geotiff_uris:
-        raise ValueError('Unable to read GeoTIFF URIs from STAC.')
-    return labels_uri, labels_box, geotiff_uris
+
+    label_items = [
+        item for item in cat.get_all_items() if 'label' in item.stac_extensions
+    ]
+
+    if len(label_items) == 0:
+        raise ValueError('Unable to find any label items in STAC catalog.')
+
+    if len(label_items) > 1:
+        raise ValueError('Found more than one label item in STAC catalog.')
```

Comment:
The exports from GroundWork will have plural label items when there's more than one project in the GroundWork campaign, so this will be pretty brittle. It might be useful for this instead for this function to take a label item ID and try to find the label item with that ID instead of relying on singularity.

Reply:
Is it correct to associate an image with a label by resolving the label item's links that have rel == "source"? Can label items have more than one such link?

Reply:
Yes that's correct. Having multiple sources [isn't explicitly forbidden](https://github.com/stac-extensions/label/#links-source-imagery) but we don't do that and it's not clear how to interpret that case to me